### PR TITLE
Use java 8 syntax to avoid IDE inspection warnings

### PR DESCRIPTION
**Pull request by anistor**
* also make some internall classes private

### CODE REVIEW

File: object-filter/src/main/java/org/infinispan/objectfilter/impl/aggregation/Grouper.java [hunk 1]
```diff
@@ -152,11 +152,6 @@ public void addRow(Object[] row) {
             private final Iterator<Object[]> iterator = groups.values().iterator();
 
             @Override
-            public void remove() {
```
Comment (anistor) [inline object-filter/src/main/java/org/infinispan/objectfilter/impl/aggregation/Grouper.java]: default impl is good enough.

File: query/src/main/java/org/infinispan/query/dsl/embedded/impl/HibernateSearchIndexedFieldProvider.java [hunk 2]
```diff
@@ -65,7 +65,7 @@ public boolean isStored(List<String> propertyPath) {
       if (propertyMetadata == null) {
          return false;
       }
-      for (DocumentFieldMetadata dfm : propertyMetadata.getFieldMetadata()) {
```
Comment (anistor) [inline query/src/main/java/org/infinispan/query/dsl/embedded/impl/HibernateSearchIndexedFieldProvider.java]: avoid using a deprecated method.

**Comment by gustavonalle** [comment]
Is CI gone?

**Comment by gustavonalle** [comment]
"JVM crashed" on CI, but it doesn't seem related to this PR, any ideas?

**Comment by anistor** [comment]
I tried re-triggering the build but got same error. Other PRs fail similarly.

**Comment by anistor** [comment]
A  rare alignment of planet in the early morning sky: all tests are green. :four_leaf_clover: :four_leaf_clover: :four_leaf_clover::four_leaf_clover:

**Comment by slaskawi** [comment]
Pulling

**Comment by slaskawi** [comment]
Integrated thanks!

**Comment by anistor** [comment]
Thanks!

**Pull request closed without merging**